ddtrace/tracer/spancontext.go

```diff
@@ -217,21 +217,44 @@ func (t *trace) setTag(key, value string) {
 	t.tags[key] = value
 }
 
-func (t *trace) setSamplingPriorityLocked(service string, p int, sampler samplernames.SamplerName, rate float64) {
+func (t *trace) setPropagatingTag(key, value string) {
+	if t.propagatingTags == nil {
+		t.propagatingTags = make(map[string]string, 1)
+	}
+	t.propagatingTags[key] = value
+}
+
+func (t *trace) setSamplingPriorityLocked(service string, p int, sampler samplernames.SamplerName, rate float64, span *span) {
 	if t.locked {
 		return
 	}
 	if t.priority == nil {
 		t.priority = new(float64)
 	}
 	*t.priority = float64(p)
-	if sampler != samplernames.Upstream {
-		if t.upstreamServices != "" {
-			t.setTag(keyUpstreamServices, t.upstreamServices+";"+compactUpstreamServices(service, p, sampler, rate))
-		} else {
-			t.setTag(keyUpstreamServices, compactUpstreamServices(service, p, sampler, rate))
-		}
+	_, ok := t.propagatingTags[keyDecisionMaker]
+	if p > 0 && !ok {
+		t.setServiceDecisionMaker(service, sampler, span)
+	}
+	if p <= 0 && ok {
+		delete(t.propagatingTags, keyDecisionMaker)
```

Comment:
```suggestion
		// a negative priority removes a decision maker
		delete(t.propagatingTags, keyDecisionMaker)
```
Why are we not setting the decision maker for `p <= 0`?

Reply:
Right, but why don't we need a decision maker if p<=0. That's a decision too, and it will reach the agent. Or is this decision maker only used by the backend? In which case it makes sense, and we should comment that.

Reply:
The decision maker is primarily for the backend so that it can highlight _why_ a particular span was kept. This is mostly in support of the a better ingestion control page so customers can better see where their spans are coming from and change the sampling rules in those services if desired. @Barbayar - feel free to correct me here or provide some additional context if needed!

Reply:
Yes, it's used by only on backend. There are two reasons.
- Highlighting the span made the decision on the UI, so customer knows that which span caused the sampling.
- Correlate hash and service. We can build the hash table on the backend from services we've seen. But, The hash tracers compute are based on unnormalized service names, but the hashes backend computes on the backend will be based on normalized service names. So, we will build a fallback table using this tag.

Reply:
Thanks for explaining 🙇🏻